File: spark-jobs/py_equivalents/etl_monitor_reporter.py
# -*- coding: utf-8 -*-
import time
from loguru import logger

def record_job_history(job_name, status, start_ts, end_ts, input_rows=0, output_rows=0, error_msg=""):
    duration_ms = int((end_ts - start_ts) * 1000)
    logger.info(f"[ETL Monitor] Job: {job_name} | Status: {status} | Duration: {duration_ms}ms | Input: {input_rows} | Output: {output_rows}")

File: spark-jobs/src/main/scala/com/mo/medgraph/etl/ETLMonitorReporter.scala
package com.mo.medgraph.etl

object ETLMonitorReporter {
  def report(jobName: String, status: String, durationMs: Long, inputRows: Long, outputRows: Long): Unit = {
    println(s"[ETL Monitor] Job: $jobName | Status: $status | Duration: ${durationMs}ms | In: $inputRows | Out: $outputRows")
  }
}
